### PR TITLE
Unit entity state transitions (SEIGED -> TRANSITION -> MOVE -> TRANSITION -> SEIGED)

### PR DESCRIPTION
**Pull request by kamikaz1k**
Using the `xstate` library to make the state machine. TBH I learned the library while trying to hack together the machine. Looking at the docs, it is probably not exactly correct, but I at least it is very clear what is managed by the state machine and what is managed by the entity directly.

Also, this machine doesn't yet handle queueing moveTo commands while unit is transitioning to seige mode.

Library site: https://xstate.js.org/

### CODE REVIEW

File: src/entities/Unit.ts [hunk 1]
```diff
@@ -217,6 +260,7 @@ function moveTowardsTarget(unit: Unit, delta: number) {
     const isAtActiveTarget = distance <= UNIT_SNAP_DISTANCE;
 
     if (isAtActiveTarget) {
+      // Pops first element from array, essentially marking point as consumed
```
Comment (nafeu) [inline src/entities/Unit.ts:263]: This comment is fine for now but I'm probably going to remove it later in favour of named anonymous functions ie:

```javascript
(function removeFirstCoordinateFromActivePath(){
  unit.activePath.shift();
})();
```

**Review by nafeu — APPROVED** [review]
Code looks great, also checked out the branch and tried out it out locally and works fine. Great stuff man! The statemachine is awesome

File: src/entities/Unit.ts (hunk 1 above)
Comment (kamikaz1k) [reply src/entities/Unit.ts:263]: Seems like unnecessary overhead to wrap it in a function? why not just leave the comment in?

Or, make a proper Path class that has method for interacting with it?

**Comment by kamikaz1k** [comment]
rebased from main, should be identical...

can do follow ups if I screwed up though

**Pull request merged**